### PR TITLE
Bump to Submariner 0.15

### PR DESCRIPTION
**Pull request by skitt**
This requires a number of updates in cloud-prepare-based code, and a bump to Ginkgo v2.

Depends on https://github.com/openshift/release/pull/36750
Depends on https://github.com/openshift/release/pull/36803

### CODE REVIEW

File: pkg/cloud/manifestwork/machineset.go
Comment (skitt) [inline pkg/cloud/manifestwork/machineset.go:148]: Oops, I need to handle this.

**Comment by skitt** [comment]
This requires Go 1.19 builders.

**Comment by skitt** [comment]
/cc @sridhargaddam @tpantelis 

This is blocked by Sonar’s coverage requirements. I tried adding some (useless) tests, and the coverage percentage went _down_! In any case this is all cloud-provider-specific code, and UTs don’t seem particularly useful there. If you’re OK with this, I’ll force-merge after the PR is approved.

**Review by tpantelis — APPROVED** [review]

**Pull request merged**
